### DISCUSSION TITLE
Ability to group topics in 3D panel

### DISCUSSION THREAD

**Discussion by tonynajjar** [foxglove/community#677] [Q&A]
Something I miss from Rviz is the ability to group topics together for easier readability (e.g Navigation, Perception, etc...). 
It's especially useful when you don't know which particular topic you're looking for but want to explore all topics in a certain group e.g. Navigation, to debug an issue related to that group

It would be nice if for the 3D panel we could define Groups (folders basically); if the topic exists, it's put automatically in that group

**Comment by defunctzombie** [comment]
How would you like to define the group?

**Reply by tonynajjar** [reply]
For me personally, it's only important that I can do so in the 3D panel json config. I guess other people would also like some interface where you can create a folder and drag topics into it. I think such design decisions can be taken by the Foxglove team, I have no preference

**Comment by snosenzo** [comment]
> useful when trying to find a topic

just want to plug that there is a search/filter bar at the top of the 3D panel settings as well which allows you to filter all settings in the panel including individual topics. 
<img width="319" alt="image" src="https://user-images.githubusercontent.com/10187776/229171370-23782bfc-19af-46bd-ad8d-6d691d296c6a.png">

**Reply by tonynajjar** [reply]
Yep thanks for pointing that out I know about it and use it sometimes but Groups would still be useful in the scenario where we don't know the name of the topic.
E.g. I see in my 3D visualization that the robot is navigating weirdly and I'd like to explore all the topics in a Navigation group to see if something there could help me debug the issue.

**Reply by tonynajjar** [reply]
Updated the original post
